### ISSUE TITLE
feat(spinnaker): Support dynamic config (Spring cloud config) across all services

### ISSUE THREAD

**Issue by jasonmcintosh** [spinnaker/spinnaker#6799]
### Issue Summary:
Currently some of the services support externalized configuration via an SCC system.  This is done via 
https://github.com/spinnaker/kork/blob/master/kork-cloud-config-server/src/main/java/com/netflix/spinnaker/kork/configserver/ConfigServerBootstrap.java
which sets up some default properties for the project related to SCC & also force includes libraries to support this

Currently this is somewhat haphazardly implemented across the services.  Currently, config-server is not added automatically as a runtime dependency.  Additionally, startup configs are not always implementing the required code.  

Example:
https://github.com/spinnaker/kork/blob/master/kork-config/src/main/java/com/netflix/spinnaker/kork/boot/DefaultPropertiesBuilder.java
https://github.com/spinnaker/kork/blob/master/kork-cloud-config-server/src/main/java/com/netflix/spinnaker/kork/configserver/ConfigServerBootstrap.java
BOTH set system properties and both are required via:
https://github.com/spinnaker/clouddriver/blob/master/clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/Main.groovy#L75

Related spinnaker issues/PRs:
https://github.com/spinnaker/spinnaker/issues/6780

### Description:
Looking at cleaning this up.  Specifically:

- [ ] Merge ConfigServerBootstrap config into DefaultPropertiesBuilder so we have ONE place to set "standard" startup config properties.  This would involve changing main startup slightly to pass the app name
- [ ] Include kork-cloud-config-server as an runtime dep.  This will automatically add all the SCC libs to all the projects at least as a runtime dependency.  This should work fine since we'd move the COnfigServerBootstrap into the default properties builder
- [ ] Make sure all spinnaker "Main" classes support this.  Probably a "SpinnakerMain" or similar generic bootstrap everything can extend from that does things like this.  NOTE:  This will be MUCH easier with the monorepo plans.
- [ ] Apply this to Orca
- [ ] Apply this to Kayenta
- [ ] Apply this to Igor
- [ ] Apply this to Gate
- [ ] Apply this to Front50
- [ ] Apply this to Fiat
- [ ] Apply this to Rosco
- [ ] Apply this to Clouddriver

### Additional Details:
* INCLUDED: I'm not sure `defaultSystemProperty("spring.application.name", applicationName);` is needed.  Part of this would remove that and make sure it's NOT needed (fixing either base properties or other settings to make sure it IS set appropriately in various files).
* CAUTION/INCLUDED: Make sure clouddriver isn't ALWAYS "clouddriver" - aka clouddriver-caching HAS to be a supported option for different configurations.  Similarly for echo and/or other services allowing custom names to be set as needed for services.
* NOT part of this:  Eureka (aka archaius) is still used as a lib.  Separate point but not sure this is needed, and is an alternative to this.
* NOT part of this:  Changing any config properties or fixing "oddities" that have been tweaked in the project for viable spring alternatives.

**Comment by dbyron-sf** [comment]
At first glance this feels like a good direction.  For this specific item though:

> * Merge ConfigServerBootstrap config into DefaultPropertiesBuilder so we have ONE place to set "standard" startup config properties.  This would involve changing main startup slightly to pass the app name

I think I'd rather go in the other direction...DefaultPropertiesBuilder is in kork-config, and ConfigServerBootstrap is in kork-cloud-config-server.  It feels better for kork-cloud-config-server to depend on kork-config than the other direction.

**Comment by jasonmcintosh** [comment]
Good call :)

**Comment by jasonmcintosh** [comment]
Plan for a refactor/rename.  Wish we had that mono repo going ;) Make this a lot easier as a single PR!

**Comment by j-sandy** [comment]
Agreed with merge approach suggested by @dbyron-sf. And it is viable to check the adoption of dynamic config for specific features by each spinnaker service on requirement basis, like the adoption of dynamic accounts in clouddriver.

**Comment by jasonmcintosh** [comment]
Going to close this ticket.  At this point... we're modifying most resources to be account storage aware, which negates the need for much of this type of dynamic configuration approach.
